contracts/controllers/DelegatedController.sol

```diff
@@ -0,0 +1,187 @@
+    }
+
+    function setOneSplit(address _onesplit) external {
+        require(msg.sender == governance, "!governance");
+        onesplit = _onesplit;
+    }
+
+    function setGovernance(address _governance) external {
+        require(msg.sender == governance, "!governance");
+        governance = _governance;
+    }
+
+    function setConverter(
+        address _input,
+        address _output,
+        address _converter
+    ) external {
+        require(msg.sender == governance, "!governance");
+        converters[_input][_output] = _converter;
+    }
+
+    function setStrategy(address _vault, address _strategy) external {
+        require(msg.sender == governance, "!governance");
+        address _current = strategies[_vault];
+        if (_current != address(0)) {
+            Strategy(_current).withdrawAll();
+        }
+        strategies[_vault] = _strategy;
+        isStrategy[_strategy] = true;
+        vaults[_strategy] = _vault;
+        isVault[_vault] = true;
+    }
+
+    function want(address _vault) external view returns (address) {
+        return Strategy(strategies[_vault]).want();
+    }
+
+    function earn(address _vault, uint256 _amount) public {
+        address _strategy = strategies[_vault];
+        address _want = Strategy(_strategy).want();
+        IERC20(_want).safeTransfer(_strategy, _amount);
+        Strategy(_strategy).deposit();
+    }
+
+    function balanceOf(address _vault) external view returns (uint256) {
+        return Strategy(strategies[_vault]).balanceOf();
+    }
+
+    function withdrawAll(address _strategy) external {
+        require(msg.sender == governance, "!governance");
+        // WithdrawAll sends 'want' to 'vault'
+        Strategy(_strategy).withdrawAll();
+    }
+
+    function inCaseTokensGetStuck(address _token, uint256 _amount) external {
+        require(msg.sender == governance, "!governance");
+        IERC20(_token).safeTransfer(governance, _amount);
+    }
+
+    function inCaseStrategyGetStruck(address _strategy, address _token) external {
+        require(msg.sender == governance, "!governance");
+        Strategy(_strategy).withdraw(_token);
+        IERC20(_token).safeTransfer(governance, IERC20(_token).balanceOf(address(this)));
+    }
+
+    function getExpectedReturn(
+        address _strategy,
+        address _token,
+        uint256 parts
+    ) external view returns (uint256 expected) {
+        uint256 _balance = IERC20(_token).balanceOf(_strategy);
+        address _want = Strategy(_strategy).want();
+        (expected, ) = OneSplitAudit(onesplit).getExpectedReturn(_token, _want, _balance, parts, 0);
+    }
+
+    function claimInsurance(address _vault) external {
+        require(msg.sender == governance, "!governance");
+        DelegatedVault(_vault).claimInsurance();
+    }
+
+    // Only allows to withdraw non-core strategy tokens ~ this is over and above normal yield
+    function delegatedHarvest(address _strategy, uint256 parts) external {
```

Comment:
This is... a lot to digest. Will have to take a second look to see if I fully understand it.